compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3084,11 +3094,11 @@ object Parsers {
         else if (isDefIntro(localModifierTokens))
           if (closureMods.contains(in.token)) {
             val start = in.offset
-            var imods = modifiers(closureMods)
+            var imods = modifiers(closureMods + IMPLIED)
```

Comment:
I believe this is all unnecessary. `IMPLIED` is not in closure mods. So the only way you could recognize an `IMPLIED` here is if it follows a closure mod. Something like
```
implicit implied match
given implied match
eraded implied match
```
But these are all illegal, so it can only be wrong if we recognize an IMPLIED here.